BeerDispancer/BeerDispencer.Infrastructure/Persistence/BeerDispencerUOW.cs

```diff
@@ -55,6 +54,35 @@ public TransactionScope StartTransaction()
             return _transaction;
         }
 
+        public async Task ProcessPaymentAsync(Guid dispencerId, double amount)
+        {
+
+            using var transaction = StartTransaction();
+
+            var payment = new Payments
+            {
+                DispencerId = dispencerId,
+                Amount = amount,
+                Status = PaymentStatus.Completed,
+                CreatedAt = DateTime.UtcNow,
+                UpdatedAt = DateTime.UtcNow
+            };
+
+            var outbox = new Outbox
+            {
+                Data = JsonConvert.SerializeObject(payment),
+                MessageType =nameof(Payments),
+                CreatedAt = DateTime.UtcNow,
+                Status = OperationStatus.Created
+            };
+
+            await _dbcontext.Payments.AddAsync(payment);
+
+            await _dbcontext.Outbox.AddAsync(outbox);
+            await _dbcontext.SaveChangesAsync(_cancellationToken);
+            CommitTransaction();
+        }
+
         public void CommitTransaction()
         {
```

Comment:
Why not to dispatch here?

Reply:
Check Domain events pattern -> https://microservices.io/patterns/data/domain-event.html